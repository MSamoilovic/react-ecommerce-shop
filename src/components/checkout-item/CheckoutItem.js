import React from "react";
import "./CheckoutItem.scss";
import { connect } from "react-redux";
import {
  clearItemFromCart,
  removeItemsFromCheckout,
  addItem,
} from "../../redux/cart/cart-actions";

const mapDispatchToProps = (dispatch) => ({
  clearItem: (item) => dispatch(clearItemFromCart(item)),
  removeItem: (item) => dispatch(removeItemsFromCheckout(item)),
  addItem: (item) => dispatch(addItem(item)),
});

export const CheckoutItem = ({ item, clearItem, removeItem, addItem }) => {
  const { name, price, quantity, imageUrl } = item;
  return (
    <div className="checkout-item">
      <div className="image-container">
        <img src={imageUrl} alt="productImage" />
      </div>
      <span className="name">{name}</span>
      <span className="quantity">
        <div className="arrow" onClick={() => removeItem(item)}>
          &#10094;
        </div>
        <span className="value">{quantity}</span>
        <div className="arrow" onClick={() => addItem(item)}>
          &#10095;
        </div>
      </span>
      <span className="price">${price}</span>
      <div className="remove-button" onClick={() => clearItem(item)}>
        <span role="img" aria-label="Remove Button">
          &#x274C;
        </span>
      </div>
    </div>
  );
};

export default connect(null, mapDispatchToProps)(CheckoutItem);
