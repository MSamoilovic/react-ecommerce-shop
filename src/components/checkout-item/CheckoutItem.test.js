import React from "react";
import { shallow } from "enzyme";
import { CheckoutItem } from "./CheckoutItem";
import logger from "redux-logger";

describe("Checkout Item component", () => {
  let wrapper;
  let mockClearItem;
  let mockRemoveItem;
  let mockAddItem;

  beforeEach(() => {
    mockClearItem = jest.fn();
    mockRemoveItem = jest.fn();
    mockAddItem = jest.fn();

    const mockProps = {
      item: {
        imageUrl: "www.testImage.com",
        price: 1,
        quanity: 1,
        name: "hat",
      },
      clearItem: mockClearItem,
      removeItem: mockRemoveItem,
      addItem: mockAddItem,
    };

    wrapper = shallow(<CheckoutItem {...mockProps} />);
  });

  it("should render", () => {
    //console.log(wrapper.debug())  
    expect(wrapper).toMatchSnapshot();
  });

  it('should call clearItem when remove button is clicked', ()=> {
      wrapper.find('.remove-button').simulate('click')
      expect(mockClearItem).toHaveBeenCalled()
  })

  it('should remove item when left arrow is clicked', ()=> {
      wrapper.find('.quantity').childAt(0).simulate('click')
      expect(mockRemoveItem).toHaveBeenCalled()
  })

  it('should add item when right arrow is clicked', ()=> {
      wrapper.find('.quantity').childAt(2).simulate('click')
      expect(mockAddItem).toHaveBeenCalled()
  })
});
