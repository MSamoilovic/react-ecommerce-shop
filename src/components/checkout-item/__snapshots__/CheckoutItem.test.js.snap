// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout Item component should render 1`] = `ShallowWrapper {}`;
